test/PushNotificationTests/TestFunctions.cpp

```diff
@@ -0,0 +1,151 @@
+    if (status != PushNotificationChannelStatus::CompletedSuccess)
+    {
+        return result.ExtendedError(); // did not produce a channel
+    }
+
+    result.Channel().Close();
+    return S_OK;
+}
+
+void PushNotificationTestFunctions::ChannelRequestUsingNullRemoteId()
+{
+    VERIFY_THROWS_HR(PushNotificationManager::Default().CreateChannelAsync(winrt::guid()).get(), E_INVALIDARG);
+}
+
+void PushNotificationTestFunctions::ChannelRequestUsingRemoteId()
+{
+    auto channelOperation{ PushNotificationManager::Default().CreateChannelAsync(c_azureRemoteId) };
+    VERIFY_SUCCEEDED(ChannelRequestHelper(channelOperation));
+}
+
+// Currently failing.
+void PushNotificationTestFunctions::MultipleChannelClose()
+{
+    auto channelOperation{ PushNotificationManager::Default().CreateChannelAsync(c_azureRemoteId) };
+    if (channelOperation.wait_for(c_timeout) != winrt::Windows::Foundation::AsyncStatus::Completed)
+    {
+        channelOperation.Cancel();
+        VERIFY_FAIL(L"Channel request hit timeout.");
+    }
+
+    auto result{ channelOperation.GetResults() };
+    VERIFY_ARE_EQUAL(result.Status(), PushNotificationChannelStatus::CompletedSuccess);
+
+    result.Channel().Close();
+    VERIFY_THROWS_HR(result.Channel().Close(), WPN_E_CHANNEL_CLOSED);
+}
+
+void PushNotificationTestFunctions::VerifyRegisterAndUnregister()
+{
+    auto pushNotificationManager{ PushNotificationManager::Default() };
+    pushNotificationManager.Register();
+    pushNotificationManager.Unregister();
+}
+
+void PushNotificationTestFunctions::VerifyRegisterAndUnregisterAll()
+{
+    auto pushNotificationManager{ PushNotificationManager::Default() };
+    pushNotificationManager.Register();
+    pushNotificationManager.UnregisterAll();
+}
+
+void PushNotificationTestFunctions::MultipleRegister()
+{
+    auto pushNotificationManager{ PushNotificationManager::Default() };
+    pushNotificationManager.Register();
+    VERIFY_THROWS_HR(pushNotificationManager.Register(), HRESULT_FROM_WIN32(ERROR_ALREADY_REGISTERED));
+}
+
+void PushNotificationTestFunctions::VerifyMultipleRegisterAndUnregister()
+{
+    PushNotificationManager::Default().Register();
+    PushNotificationManager::Default().Unregister();
+    VERIFY_THROWS_HR(PushNotificationManager::Default().Register(), HRESULT_FROM_WIN32(ERROR_ALREADY_REGISTERED));
+}
+
+void PushNotificationTestFunctions::VerifyMultipleRegisterAndUnregisterAll()
+{
+    PushNotificationManager::Default().Register();
+    PushNotificationManager::Default().UnregisterAll();
+
+    PushNotificationManager::Default().Register();
+    PushNotificationManager::Default().UnregisterAll();
+}
+
+void PushNotificationTestFunctions::VerifyUnregisterTwice()
+{
+    PushNotificationManager::Default().Register();
+    PushNotificationManager::Default().Unregister();
+    VERIFY_THROWS_HR(PushNotificationManager::Default().Unregister(), E_UNEXPECTED);
+}
+
+void PushNotificationTestFunctions::VerifyUnregisterAll()
```

Comment:
What is the difference between VerifyRegisterAndUnregisterAll and this test?

Reply:
There is no difference. Removing.